### PR TITLE
Rewrote Search and Gallery Item form using MVP

### PR DESCRIPTION
**Pull request by aurisgo**

### CODE REVIEW

File: RoadRangerBackEnd/Report/MailReportSender.cs [hunk 1]
```diff
@@ -27,14 +28,21 @@ public void SendGeneretedMail(Car car)
...
+                    From = mailAddress    //Mūsų gmail laikinas, iš kurio siunčia
+                };
+                mail.To.Add(defaultRecipient);     //adresas iš textbox
+                mail.Subject = subject;     //Tema
+                mail.Body = body;           //Tekstas
+                smtpServer.Send(mail);      //siunčiam
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
```
Comment (modestas-jucevicius) [inline RoadRangerBackEnd/Report/MailReportSender.cs]: Maybe show the message in the app with some kind of pop up would be better than Console logging it

File: RoadRangerMobileApp/RoadRangerMobileApp.Android/RoadRangerMobileApp.Android.csproj.user [hunk 2]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <AndroidDesignerPreferredTheme>Theme.DeviceDefault.NoActionBar.Fullscreen</AndroidDesignerPreferredTheme>
     <SelectedDevice>Android_Accelerated_x86_Oreo</SelectedDevice>
-    <SelectedDevice>Android_Accelerated_x86_Oreo</SelectedDevice>
-    <DefaultDevice>Android_Accelerated_x86_Oreo</DefaultDevice>
+    <SelectedDevice>Android_Accelerated_x86_Nougat</SelectedDevice>
+    <DefaultDevice>Android_Accelerated_x86_Nougat</DefaultDevice>
```
Comment (modestas-jucevicius) [inline RoadRangerMobileApp/RoadRangerMobileApp.Android/RoadRangerMobileApp.Android.csproj.user:7]: Why change default?

File: RoadRangerMobileApp/RoadRangerMobileApp/Presenters/SearchItemPresenter.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+using RoadRangerMobileApp.Views;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace RoadRangerMobileApp.Presenters
+{
+    public class SearchItemPresenter
+    {
+        protected ICapturedCarFinder finder = new CapturedCarFinder();
```
Comment (modestas-jucevicius) [inline RoadRangerMobileApp/RoadRangerMobileApp/Presenters/SearchItemPresenter.cs]: readonly

File: RoadRangerMobileApp/RoadRangerMobileApp/Views/MainPage.xaml.cs [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
 
 namespace RoadRangerMobileApp.Views
 {
+    [XamlCompilation(XamlCompilationOptions.Compile)]
     public partial class MainPage : ContentPage
     {
         public MainPage()
         {
-            NavigationPage.SetHasNavigationBar(this, false);
+            //NavigationPage.SetHasNavigationBar(this, true);
```
Comment (modestas-jucevicius) [inline RoadRangerMobileApp/RoadRangerMobileApp/Views/MainPage.xaml.cs]: leftover redundant code?

**Review by modestas-jucevicius — DISMISSED** [review]
If you're trying to implement what the lecturer said, this is still not it as far as I understand, View shouldn't be aware of Presenter, shouldn't have this.presenter. Presenter should react to View events instead.

File: RoadRangerMobileApp/RoadRangerMobileApp/Views/MainPage.xaml.cs (hunk 4 above)
Comment (aurisgo) [reply RoadRangerMobileApp/RoadRangerMobileApp/Views/MainPage.xaml.cs]: This code line is necessary. So, I uncommented it.

File: RoadRangerMobileApp/RoadRangerMobileApp.Android/RoadRangerMobileApp.Android.csproj.user (hunk 2 above)
Comment (aurisgo) [reply RoadRangerMobileApp/RoadRangerMobileApp.Android/RoadRangerMobileApp.Android.csproj.user:7]: This changes according to my emulator device.

**Review by pauliusanuzas — APPROVED** [review]

**Pull request merged**